pkg/ccl/crosscluster/logical/logical_replication_job_test.go

```diff
@@ -648,57 +648,95 @@ func TestRandomTables(t *testing.T) {
 
-	// TODO(ssd): We have to turn off randomized_anchor_key
-	// because this, in combination of optimizer difference that
-	// might prevent CommitInBatch, could result in the replicated
-	// transaction being too large to commit.
-	runnerA.Exec(t, "SET CLUSTER SETTING kv.transaction.randomized_anchor_key.enabled=false")
+	dbAURL, cleanup := s.PGUrl(t, serverutils.DBName("a"))
+	defer cleanup()
 
-	// Workaround for the behaviour described in #127321. This
-	// ensures that we are generating rows using similar
-	// optimization decisions to our replication process.
-	runnerA.Exec(t, "SET plan_cache_mode=force_generic_plan")
+	var jobBID jobspb.JobID
+	streamStartStmt := fmt.Sprintf("CREATE LOGICAL REPLICATION STREAM FROM TABLE %[1]s ON $1 INTO TABLE %[1]s", tableName)
+	runnerB.QueryRow(t, streamStartStmt, dbAURL.String()).Scan(&jobBID)
 
-	numInserts := 20
-	_, err := randgen.PopulateTableWithRandData(rng,
-		sqlA, tableName, numInserts, nil)
-	require.NoError(t, err)
+	WaitUntilReplicatedTime(t, s.Clock().Now(), runnerB, jobBID)
+	compareReplicatedTables(t, s, "a", "b", tableName, runnerA, runnerB)
+}
 
-	addCol := fmt.Sprintf(`ALTER TABLE %s `+lwwColumnAdd, tableName)
-	runnerA.Exec(t, addCol)
-	runnerB.Exec(t, addCol)
+func TestMultipleRandomTables(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	clusterArgs := base.TestClusterArgs{
+		ServerArgs: base.TestServerArgs{
+			DefaultTestTenant: base.TestControlsTenantsExplicitly,
+			Knobs: base.TestingKnobs{
+				JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(),
+			},
+		},
+	}
+
+	rng, _ := randutil.NewPseudoRand()
+	with := false
+	maybeAddWith := func(stmt string) string {
+		if !with {
+			stmt = stmt + " WITH"
+		}
+		with = true
+		return stmt
+	}
+
+	maybeAddUDF := func(stmt, tableName string) string {
+		if rng.Intn(2) == 0 {
+			// Already added an option, add comma
+			if with {
+				stmt = stmt + ","
+			}
+			stmt = maybeAddWith(stmt) + fmt.Sprintf(" FUNCTION repl_apply FOR TABLE %s", tableName)
+		}
+		return stmt
+	}
+
+	ctx := context.Background()
+	server, s, runnerA, runnerB := setupLogicalTestServer(t, ctx, clusterArgs, 1)
+	defer server.Stopper().Stop(ctx)
+
+	tableName1 := "rand_table_1"
+	tableName2 := "rand_table_2"
+	SetupTablePair(t, ctx, s, runnerA, runnerB, tableName1)
+	SetupTablePair(t, ctx, s, runnerA, runnerB, tableName2)
 
 	dbAURL, cleanup := s.PGUrl(t, serverutils.DBName("a"))
 	defer cleanup()
 
-	streamStartStmt := fmt.Sprintf("CREATE LOGICAL REPLICATION STREAM FROM TABLE %[1]s ON $1 INTO TABLE %[1]s", tableName)
+	runnerB.Exec(t, fmt.Sprintf(testingUDFAcceptProposedBase, tableName1))
+
 	var jobBID jobspb.JobID
-	runnerB.QueryRow(t, streamStartStmt, dbAURL.String()).Scan(&jobBID)
+	streamStartStmt := fmt.Sprintf("CREATE LOGICAL REPLICATION STREAM FROM TABLES (%[1]s, %[2]s) ON $1 INTO TABLES (%[1]s, %[2]s)", tableName1, tableName2)
+	streamStartStmt = maybeAddUDF(streamStartStmt, tableName1)
```

Comment:
@msbutler I know you wanted to add UDFs for this test, but unless I can figure out a solution to https://cockroachlabs.slack.com/archives/C03JCUUSCD6/p1725923974249999, it's going to be tough

Reply:
I think we may have miscommunicated. No need to set up a UDF on this test. I did say that we should add a LWW UDF test case to the LWW test you built, but that should be done in a separate PR.

Reply:
oh great!